api/services/auth.go

```diff
@@ -66,128 +68,112 @@ type AuthService interface {
+			StatusUpdatedAt: clock.Now(),
+			Name:            strings.ToLower(hostname),
+			Identity:        &models.DeviceIdentity{MAC: req.Identity.MAC},
+			PublicKey:       req.PublicKey,
+			RemoteAddr:      req.RealIP,
+			Tags:            []string{},
+			Position:        &models.DevicePosition{Longitude: position.Longitude, Latitude: position.Latitude},
+		}
 
-	// The order here is critical as we don't want to register devices if the tenant id is invalid
-	namespace, err := s.store.NamespaceGet(ctx, device.TenantID)
-	if err != nil {
-		return nil, NewErrNamespaceNotFound(device.TenantID, err)
-	}
+		if req.Info != nil {
+			device.Info = &models.DeviceInfo{
+				ID:         req.Info.ID,
+				PrettyName: req.Info.PrettyName,
+				Version:    req.Info.Version,
+				Arch:       req.Info.Arch,
+				Platform:   req.Info.Platform,
+			}
+		}
 
-	hostname = strings.ToLower(hostname)
+		if _, err := s.store.DeviceCreate(ctx, device); err != nil {
+			return nil, NewErrDeviceCreate(models.Device{}, err)
+		}
 
-	inserted, err := s.store.DeviceCreate(ctx, device, hostname)
-	if err != nil {
-		return nil, NewErrDeviceCreate(device, err)
-	}
+		if err := s.store.NamespaceIncrementDeviceCount(ctx, req.TenantID, device.Status, 1); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := s.store.DeviceUpdate(ctx, req.TenantID, uid, &models.DeviceChanges{LastSeen: clock.Now(), DisconnectedAt: nil}); err != nil {
+			log.WithError(err).Error("failed to updated device to online")
 
-	if inserted {
-		if err := s.store.NamespaceIncrementDeviceCount(ctx, req.TenantID, models.DeviceStatusPending, 1); err != nil {
 			return nil, err
 		}
 	}
 
-	for _, uid := range req.Sessions {
-		if err := s.store.SessionSetLastSeen(ctx, models.UID(uid)); err != nil {
+	for _, sessionUID := range req.Sessions {
+		if err := s.store.SessionSetLastSeen(ctx, models.UID(sessionUID)); err != nil {
+			log.WithError(err).WithField("session_uid", sessionUID).Warn("cannot set session's last seen")
+
 			continue
 		}
 	}
 
-	dev, err := s.store.DeviceResolve(ctx, store.DeviceUIDResolver, device.UID, s.store.Options().InNamespace(device.TenantID))
-	if err != nil {
-		return nil, NewErrDeviceNotFound(models.UID(device.UID), err)
-	}
-	if err := s.cache.Set(ctx, strings.Join([]string{"auth_device", key}, "/"), &Device{Name: dev.Name, Namespace: namespace.Name}, time.Second*30); err != nil {
-		return nil, err
+	cachedData["device_name"] = device.Name
+	cachedData["namespace_name"] = namespace.Name
+	if err := s.cache.Set(ctx, "auth_device/"+uid, cachedData, time.Second*30); err != nil {
+		log.WithError(err).Warn("cannot store device authentication metadata in cache")
 	}
 
-	return &models.DeviceAuthResponse{
-		UID:       key,
-		Token:     token,
-		Name:      dev.Name,
-		Namespace: namespace.Name,
-	}, nil
+	return &models.DeviceAuthResponse{UID: uid, Token: token, Name: cachedData["device_name"], Namespace: cachedData["namespace_name"]}, nil
```

Comment:
For structs with multiple fields like this, I personally find the multi-line style easier to read and maintain especially when new fields get added in the future. Would you consider keeping the multi-line formatting for clarity?